### PR TITLE
cmd: document -c/--checkpoints for `run` and `stage add`

### PR DESCRIPTION
**Pull request by pmrowla**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Will close https://github.com/iterative/dvc.org/issues/2253

### CODE REVIEW

**Review by dberenbaum — APPROVED** [review]
LGTM

File: content/docs/command-reference/run.md [hunk 1]
```diff
@@ -192,6 +192,11 @@ $ dvc run -n second_stage './another_script.sh $MYENVVAR'
 - `--outs-persist-no-cache <path>` - the same as `-outs-persist` except that
   outputs are not tracked by DVC (same as with `-O` above).
 
+- `-c <path`, `--checkpoints <path` - declare output file or directory that will
+  be handled as an intermediate
+  [checkpoint](/doc/command-reference/exp/run#checkpoints) by `dvc exp run`.
+  Implies `--no-exec`. Note that checkpoints are not supported by `dvc repro`.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/run.md:198]: May copy edit depending on https://github.com/iterative/dvc/pull/5599#issuecomment-796984046 ⌛

Comment (jorgeorpinel) [reply content/docs/command-reference/run.md:198]:
```suggestion
- `-c <path`, `--checkpoints <path>` - the same as `-o` but also marks the
  output as a [checkpoint](/doc/command-reference/exp/run#checkpoints).
  Implies `--no-exec`. This makes the stage incompatible with `dvc repro`.
```

Comment (jorgeorpinel) [reply content/docs/command-reference/run.md:198]:
```suggestion
- `-c <path`, `--checkpoints <path>` - the same as `-o` but also marks the
  output as a [checkpoint](/doc/command-reference/exp/run#checkpoints).
  Implies `--no-exec`. This makes the stage incompatible with `dvc repro`.
```

File: content/docs/command-reference/stage/add.md [hunk 2]
```diff
@@ -187,6 +187,11 @@ data science experiments.
 - `--outs-persist-no-cache <path>` - the same as `-outs-persist` except that
   outputs are not tracked by DVC (same as with `-O` above).
 
+- `-c <path`, `--checkpoints <path` - declare output file or directory that will
+  be handled as an intermediate
+  [checkpoint](/doc/command-reference/exp/run#checkpoints) by `dvc exp run`.
+  Note that checkpoints are not supported by `dvc repro`.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/stage/add.md:193]:
```suggestion
- `-c <path`, `--checkpoints <path>` - the same as `-o` but also marks the
  output as a [checkpoint](/doc/command-reference/exp/run#checkpoints).
  Implies `--no-exec`. This makes the stage incompatible with `dvc repro`.
```

**Review by jorgeorpinel — COMMENTED** [review]
Copy edits (will commit):

**Review by jorgeorpinel — APPROVED** [review]
Mergeable as soon as the core PR is released (ideally)

Or is it already available? (Not sure what "suppressed" means in Python argparse flags)

**Comment by pmrowla** [comment]
> Mergeable as soon as the core PR is released (ideally)
> 
> Or is it already available? (Not sure what "suppressed" means in Python argparse flags)

`argparse.SUPPRESSED` makes it so the option is available to be used but will not appear in usage/help text (so it's suppressed in `-h` output). It's normally used for hidden flags which may be set as defaults, but in this case we were using it to hide the experiments related options back when it was still in opt-in alpha/beta status for DVC 1.x

**Comment by jorgeorpinel** [comment]
Looks like I merged it accidentally via Restyled already 😆

**Pull request merged**